types/src/validator_signer.rs

```diff
@@ -61,12 +61,18 @@ impl<PrivateKey: SigningKey> ValidatorSigner<PrivateKey> {
     }
 }
 
-impl<PrivateKey: SigningKey + Genesis> ValidatorSigner<PrivateKey> {
+impl<PrivateKey: SigningKey + Genesis + NilSigningKey> ValidatorSigner<PrivateKey> {
```

Comment:
Again, this is only valid if we want this to have meaning outside of consensus.

There are (immediate) plans to make the consensus module generic in the parameter of its `ValidatorSigner`.

We can late-bind `ValidatorSigner` to have a parameter wich requires a `NilSigner` bound as well, in consensus only.